moshi/src/main/java/com/squareup/moshi/Moshi.java

```diff
@@ -158,7 +159,7 @@
       Set<? extends Annotation> annotations) {
     if (annotations == null) throw new NullPointerException("annotations == null");
 
-    type = canonicalize(type);
+    type = removeSubtypeWildcard(canonicalize(type));
```

Comment:
okay, next gotcha. This doesn’t canonicalize anymore. We don’t have a test to confirm that canonicalization is happening, and we need that too.

Reply:
Such a test should confirm that a `GenericArrayType` of `String` shares an adapter with `String[]`.